app/controllers/application_drafts_controller.rb

```diff
@@ -34,9 +34,10 @@ def remove_question
 
   def update
     draft_params = params.require(:draft)
-    question_params = draft_params.require(:questions_attributes).permit!
+    question_attributes = draft_params.require(:questions_attributes)
+                                      .permit!.to_h
```

Comment:
Well, when you say it "expects" a `Hash`, in this case you mean that it expects to be able to call `.blank?` and `.each` on it. It's clear that the Rails authors intended a `ActionController::Parameters` object to behave *like* a `Hash`, in that you can access its values with `#[]`, and can use it in `ActiveRecord` assignments. But it doesn't have `.blank?`, for some reason.

My alarm bells we set off because it *seemed* (but wasn't) like a thing you might do to git around a mass-assignment protection or something.

In this case the only vaguely questionable thing here is `.permit!`, but you're right that that was there before.